addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,387 @@
+
+    # Transfer
+    destination_account_id = fields.Many2one(string="To", comodel_name='account.account', help="Account to transfer to.")
+    display_currency_helper = fields.Boolean(string="Currency Conversion Helper", compute='_compute_display_currency_helper', help="Technical field used to indicate whether or not to display the tooltip informing a currency conversion will be performed with the transfer")
+
+    @api.constrains('percentage', 'action')
+    def _constraint_percentage(self):
+        for record in self:
+            if not (0.0 < record.percentage <= 100.0):
+                raise UserError(_("Percentage must be between 0 and 100"))
+            if record.percentage != 100 and record.action != 'accrual':
+                raise UserError(_("Percentage can only be set for Change Period method"))
+
+    @api.depends('percentage', 'move_line_ids')
+    def _compute_total_amount(self):
+        for record in self:
+            percentage = record.percentage or 100
+            total_amount = percentage * sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) / 100
+            if record.total_amount != total_amount:
+                record.total_amount = total_amount
+            record.account_type = 'income' if sum(record.move_line_ids.mapped('balance')) < 0 else 'expense'
+
+    @api.depends('total_amount', 'move_line_ids')
+    def _compute_percentage(self):
+        for record in self:
+            record.percentage = record.total_amount / sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) * 100
+            record.percentage = record.percentage or 100
+
+    @api.depends('destination_account_id')
+    def _compute_display_currency_helper(self):
+        for record in self:
+            record.display_currency_helper = bool(record.destination_account_id.currency_id)
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if self.env.context.get('active_model') != 'account.move.line' or not self.env.context.get('active_ids'):
+            raise UserError(_('This can only be used on journal items'))
+        move_line_ids = self.env['account.move.line'].browse(self.env.context['active_ids'])
+        res['move_line_ids'] = move_line_ids.ids
+
+        if any(move.state != 'posted' for move in move_line_ids.mapped('move_id')):
+            raise UserError(_('You can only change the period/account for posted journal items.'))
+        if any(move_line.reconciled for move_line in move_line_ids):
+            raise UserError(_('You can only change the period/account for items that are not yet reconciled.'))
+        if any(line.company_id != move_line_ids[0].company_id for line in move_line_ids):
+            raise UserError(_('All lines must be from the same company.'))
+        res['company_id'] = move_line_ids[0].company_id.id
+
+        allowed_actions = set(dict(self._fields['action'].selection))
+        if self.env.context.get('default_action'):
```

Comment:
Yes, sometimes, only one part of the wizard makes sense (in that case we use `default_action`). We want to check that even that part makes sense.

Maybe the context key is not the best, but it prevents setting two keys (one for setting the default action and one to say that only that action is valid). When only one is valid, we hide the action selection.

Reply:
Hmmm... I don't know. Actually, do we really want to hide the selection in some cases ? In the same idea, do we want to raise that, line 92? In that case, the user will just see an error message saying basically "you can't", but with no further explanation as for the reason why. Maybe it's a bit confusing. I think an easier solution would be to always show those possible actions, and only raise when trying to do something that is not allowed in the wizard. So, we still can pass a default action to the wizard, of course, but no need to treat it in any other way than a regular default. It would probably fit better the way we do it elsewhere in Odoo. It's mainly a usability issue, maybe tsb or avw can help.

Reply:
AVW was the one that asked for this